src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -622,11 +623,21 @@ private StateEditor doTraverse(State s0, RoutingRequest options, TraverseMode tr
                     }
                 }
 
+                //calculate time format based on the input file settings
+                TimeUnit selectedTimeUnit = options.genericFileConfiguration.getTimeFormat();
                 Instant aqiTimeInstant = Instant.ofEpochMilli(dataStartTime);
-                int dataQualityHour = (int) ChronoUnit.HOURS.between(aqiTimeInstant, requestInstant);
-                if (dataQualityHour >= 0) {
-                    if (dataQualityHour < genDataValuesForTime.length) {
-                        float value = genDataValuesForTime[dataQualityHour];
+                int dataQualityRequestedTime = 0;
+                if (selectedTimeUnit == TimeUnit.HOURS) {
+                    dataQualityRequestedTime = (int) ChronoUnit.HOURS.between(aqiTimeInstant, requestInstant);
+                } else if (selectedTimeUnit == TimeUnit.SECONDS) {
```

Comment:
What if time format is not defined in the config file? Do we use any of the formats as a default one?